File: two pointers/557. Reverse Words in a String III.js
`Given a string s, reverse the order of characters in each word within a sentence while still preserving whitespace and initial word order.

 

Example 1:

Input: s = "Let's take LeetCode contest"
Output: "s'teL ekat edoCteeL tsetnoc"
Example 2:

Input: s = "God Ding"
Output: "doG gniD"`;

const reverseWords = (s) => {
  let words = s.split(" ");
  let reversed_words = [];
  for (let i = 0; i < words.length; i++) {
    reversed_words.push(words[i].split("").reverse().join(""));
  }
  return reversed_words.join(" ");
};

File: two pointers/python/557. Reverse Words in a String III.py
class Solution:
    def reverseWords(self, s: str) -> str:

        s=list(s)

        left=0
        right=0

        while right<len(s):
            # find right
            while right<len(s) and s[right]!=' ':
                right+=1

            # reverse
            s[left:right]=self.reverse(s[left:right])

            # move left and right
            left=right+1
            right=left

        return ''.join(s)
        
    def reverse(self, s: str) -> str:
        left=0
        right=len(s)-1

        while left<right:
            # swap
            s[left],s[right]=s[right],s[left]
            left+=1
            right-=1
        
        return s



if __name__ == "__main__":

    print(Solution().reverseWords("Let's take LeetCode contest")) # "s'teL ekat edoCteeL tsetnoc"
